### ISSUE TITLE
fin share regression: --host no longer taken into account, is still in documentation though

### ISSUE THREAD

**Issue by lpeabody** [docksal/docksal#601]
**Description**

We need to expose a subdomain of the virtual host via ngrok. Previously you could run `fin share --host=subdomain.virtualhost.docksal`, and in [version 1.9 of the documentation](https://docs.docksal.io/en/latest/tools/ngrok/#overriding-default-host-value) it states that you can still do this. However, when I attempt to use --host to override, it does not tunnel to the proper subdomain. I believe this is a regression.

**Steps to reproduce the issue:**

1. Run `fin share --host=...`
2. Navigate to generated tunneled ngrok subdomain.

**Describe the results you received:**

Not viewing the expected page I would have received had I navigated to the host set in the --host parameter.

**Describe the results you expected:**

I should see the page I would see had I navigated to the host set in --host on my local system.

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
---------------------
COMPOSE_PROJECT_NAME_SAFE: corvette
COMPOSE_FILE:
/Users/les.peabody/.docksal/stacks/volumes-bind.yml
/Users/les.peabody/.docksal/stacks/stack-acquia.yml
/Users/les.peabody/Sites/corvette/.docksal/docksal.yml
/Users/les.peabody/Sites/corvette/.docksal/docksal-local.yml
ENV_FILE:
/Users/les.peabody/Sites/corvette/.docksal/docksal.env
/Users/les.peabody/Sites/corvette/.docksal/docksal-local.env

PROJECT_ROOT: /Users/les.peabody/Sites/corvette
DOCROOT: docroot
VIRTUAL_HOST: baxter.docksal
VIRTUAL_HOST_ALIASES: *.baxter.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32771

Docker Compose configuration
---------------------
services:
  cli:
    build:
      context: /Users/les.peabody/Sites/corvette/.docksal/services/cli
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      BLACKFIRE_CLIENT_ID: null
      BLACKFIRE_CLIENT_TOKEN: null
      DEFAULT_BUILD_THEME_PROFILE: bax_base
      DOCROOT: docroot
      HOST_GID: '1106437719'
      HOST_UID: '1729411499'
      PHP_IDE_CONFIG: serverName=baxter.docksal
      SECRET_ACAPI_EMAIL: null
      SECRET_ACAPI_KEY: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      VIRTUAL_HOST: baxter.docksal
      XDEBUG_CONFIG: idekey=PHPSTORM remote_host=192.168.64.1
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: corvette_cli
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - /home/docker
    - project_root:/var/www:rw,nocopy
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: null
      MYSQL_DATABASE: default
      MYSQL_INITDB_SKIP_TZINFO: null
      MYSQL_ONETIME_PASSWORD: null
      MYSQL_PASSWORD: user
      MYSQL_RANDOM_ROOT_PASSWORD: null
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.2-mysql-5.6
    ports:
    - 3306/tcp
    volumes:
    - project_root:/var/www:ro,nocopy
  mail:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: mail
    image: mailhog/mailhog
    labels:
      io.docksal.virtual-host: mail.baxter.docksal,mail.baxter.docksal.*
      io.docksal.virtual-port: '8025'
    volumes:
    - project_root:/var/www:ro,nocopy
  memcached:
    command:
    - -m
    - '128'
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: memcached
    image: memcached:1.4-alpine
  solr:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: solr
    image: docksal/solr:1.0-solr4
    labels:
      io.docksal.virtual-host: solr.baxter.docksal,solr.baxter.docksal.*
      io.docksal.virtual-port: '8983'
    volumes:
    - project_root:/var/www:ro,nocopy
  varnish:
    depends_on:
      web:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      VARNISH_BACKEND_HOST: web
    hostname: varnish
    image: docksal/varnish:1.1-varnish5
    labels:
      io.docksal.virtual-host: varnish.com.baxter.docksal,varnish.edu.baxter.docksal,varnish.prod.baxter.docksal
    volumes:
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.project-root: /Users/les.peabody/Sites/corvette
      io.docksal.virtual-host: baxter.docksal,*.baxter.docksal,www.baxter.com
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /Users/les.peabody/Sites/corvette
      o: bind
      type: none

---------------------

```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
███  OS
Darwin Mac OS X 10.12.6
Darwin BOSGEN-NMX0813.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
Mode: VirtualBox VM

███  FIN
fin version: 1.60.2

███  ENVIRONMENT
DOCKER_HOST		| tcp://192.168.64.100:2376
------------------------------------------------------------
DOCKSAL_NFS_PATH	| 

███  DOCKER
EXPECTED VERSION: 18.03.1-ce

Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:23:03 2018
  OS/Arch:      linux/amd64
  Experimental: false

███  DOCKER INFO
Containers: 25
 Running: 10
 Paused: 0
 Stopped: 15
Images: 24
Server Version: 18.03.1-ce
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 393
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.93-boot2docker
Operating System: Boot2Docker 18.03.1-ce (TCL 8.2.1); HEAD : cb77972 - Thu Apr 26 16:40:36 UTC 2018
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 3.858GiB
Name: docksal
ID: DMES:V4PS:RVEM:QWOI:VIPN:CQV3:D2HD:PFKE:ZVKA:O3CX:4ROX:CMMC
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: lpeabody
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

███  DOCKER COMPOSE
EXPECTED VERSION: 1.21.1
docker-compose version 1.21.1, build 5a3f1a3
docker-py version: 3.3.0
CPython version: 3.6.4
OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018

███  DOCKER MACHINE
EXPECTED VERSION: 0.14.0
docker-machine version 0.14.0, build 89b8332

NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
default   -        virtualbox   Stopped                                       Unknown       
docksal   *        virtualbox   Running   tcp://192.168.64.100:2376           v18.03.1-ce   

███  DOCKER: IMAGES
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
corvette_cli          latest              2ecc4c95fb61        3 hours ago         1.06GB
<none>                <none>              5fd0c2c10b6e        4 hours ago         1.05GB
docksal/cli           2.2-php7.1          f8397b8a269f        5 days ago          867MB
docksal/cli           latest              7c0c3a119c9d        11 days ago         880MB
docksal/db            1.2-mysql-5.6       0f0acba4c6d1        2 months ago        256MB
docksal/cli           2.1-php7.1          4c7331d4cf77        2 months ago        950MB
docksal/cli           <none>              b1ca4fa0852d        2 months ago        876MB
corvettetest_cli      latest              2fe19472a507        2 months ago        1.01GB
docksal/cli           2.0-php7.1          9ba7c35bb83f        2 months ago        914MB
docksal/cli           2.0-php5.6          0fea463bfbd9        2 months ago        887MB
docksal/vhost-proxy   1.2                 a1c980537985        2 months ago        166MB
wernight/ngrok        latest              8b0a114c44cf        3 months ago        27.1MB
lpeabody/blt-cli      php7-1.3            5d6f9bb46e23        3 months ago        904MB
docksal/varnish       1.1-varnish5        967b76b88b85        3 months ago        107MB
docksal/cli           1.3-php7            19bdddfeab89        8 months ago        759MB
docksal/web           2.1-apache2.4       32c1d32eba6d        9 months ago        87.5MB
docksal/vhost-proxy   1.1                 898fb21e7f2d        10 months ago       158MB
docksal/db            1.1-mysql-5.6       6feb79af9597        10 months ago       298MB
memcached             1.4-alpine          273374b463b8        10 months ago       7.68MB
mailhog/mailhog       latest              669347153085        14 months ago       19.1MB
docksal/dns           1.0                 20c7a535479d        16 months ago       61.6MB
docksal/ssh-agent     1.0                 148220cc033d        16 months ago       12.2MB
docksal/solr          1.0-solr4           fa6b236caf06        17 months ago       546MB
docksal/varnish       1.0-varnish4        35dae6b91ff9        17 months ago       99.5MB

███  DOCKER: CONTAINERS
CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS                 PORTS                                                    NAMES
dc09653e057b        docksal/varnish:1.1-varnish5   "/opt/startup.sh"        2 hours ago         Up 2 hours             80/tcp, 6082/tcp                                         corvette_varnish_1
49407fe66ff6        docksal/web:2.1-apache2.4      "httpd-foreground"       2 hours ago         Up 2 hours             80/tcp, 443/tcp                                          corvette_web_1
42b099f61585        corvette_cli                   "/opt/startup.sh sup…"   2 hours ago         Up 2 hours (healthy)   22/tcp, 3000/tcp, 9000/tcp                               corvette_cli_1
ef35c01fb77a        docksal/db:1.2-mysql-5.6       "/entrypoint.sh mysq…"   2 hours ago         Up 2 hours             0.0.0.0:32771->3306/tcp                                  corvette_db_1
62472871e629        mailhog/mailhog                "MailHog"                2 hours ago         Up 2 hours             1025/tcp, 8025/tcp                                       corvette_mail_1
536f6844357a        memcached:1.4-alpine           "docker-entrypoint.s…"   2 hours ago         Up 2 hours             11211/tcp                                                corvette_memcached_1
80ff40407397        docksal/solr:1.0-solr4         "/opt/solr/bin/solr …"   2 hours ago         Up 2 hours             8983/tcp                                                 corvette_solr_1
e7068afca4c9        docksal/ssh-agent:1.0          "/run.sh ssh-agent"      4 hours ago         Up 4 hours                                                                      docksal-ssh-agent
50cc1029819f        docksal/dns:1.0                "/opt/entrypoint.sh …"   4 hours ago         Up 4 hours             192.168.64.100:53->53/udp                                docksal-dns
340be2a85d41        docksal/vhost-proxy:1.2        "docker-entrypoint.s…"   4 hours ago         Up 4 hours             192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy

███  VIRTUALBOX
EXPECTED VERSION: 5.2.2
5.1.28r117968

███  VIRTUALBOX NETWORK INTERFACES
Name:            vboxnet0
GUID:            786f6276-656e-4074-8000-0a0027000000
DHCP:            Disabled
IPAddress:       192.168.99.1
NetworkMask:     255.255.255.0
IPV6Address:     
IPV6NetworkMaskPrefixLength: 0
HardwareAddress: 0a:00:27:00:00:00
MediumType:      Ethernet
Status:          Down
VBoxNetworkName: HostInterfaceNetworking-vboxnet0

Name:            vboxnet1
GUID:            786f6276-656e-4174-8000-0a0027000001
DHCP:            Disabled
IPAddress:       192.168.64.1
NetworkMask:     255.255.255.0
IPV6Address:     
IPV6NetworkMaskPrefixLength: 0
HardwareAddress: 0a:00:27:00:00:01
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-vboxnet1

Name:            vboxnet2
GUID:            786f6276-656e-4274-8000-0a0027000002
DHCP:            Disabled
IPAddress:       192.168.25.1
NetworkMask:     255.255.255.0
IPV6Address:     
IPV6NetworkMaskPrefixLength: 0
HardwareAddress: 0a:00:27:00:00:02
MediumType:      Ethernet
Status:          Down
VBoxNetworkName: HostInterfaceNetworking-vboxnet2

███  NFS EXPORTS

# <ds-nfs docksal
/Users 127.0.0.1 192.168.64.100 -alldirs -maproot=0:0
# ds-nfs>

███  DOCKSAL MOUNTS
192.168.64.1:/Users on /Users type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)
192.168.64.1:/Users/les.peabody/Sites/corvette on /mnt/sda1/var/lib/docker/volumes/corvette_project_root/_data type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)

███  HDD Usage
Filesystem                Size      Used Available Use% Mounted on
/dev/sda1                46.3G     14.9G     29.0G  34% /mnt/sda1

███  RAM Usage
             total       used       free     shared    buffers     cached
Mem:       4045520    3906812     138708     257272     127136    1927668
-/+ buffers/cache:    1852008    2193512
Swap:      1910088      28664    1881424

███  RAM Usage per container
CONTAINER ID        NAME                   CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
dc09653e057b        corvette_varnish_1     1.30%               7.684MiB / 3.858GiB   0.19%               29.6kB / 0B         23.8MB / 422kB      219
49407fe66ff6        corvette_web_1         0.15%               11.16MiB / 3.858GiB   0.28%               2.65MB / 3.67MB     3.96MB / 0B         109
42b099f61585        corvette_cli_1         0.03%               275.7MiB / 3.858GiB   6.98%               304MB / 143MB       34.6MB / 82.1MB     6
ef35c01fb77a        corvette_db_1          0.07%               815.8MiB / 3.858GiB   20.65%              778MB / 420MB       40.1MB / 5.53GB     25
62472871e629        corvette_mail_1        0.02%               1.918MiB / 3.858GiB   0.05%               30.7kB / 0B         7.8MB / 0B          4
536f6844357a        corvette_memcached_1   0.01%               904KiB / 3.858GiB     0.02%               30.9kB / 0B         1.18MB / 0B         7
80ff40407397        corvette_solr_1        0.19%               222.5MiB / 3.858GiB   5.63%               31kB / 0B           25.3MB / 81.9kB     37
e7068afca4c9        docksal-ssh-agent      0.00%               1000KiB / 3.858GiB    0.02%               3.18kB / 0B         2.74MB / 0B         2
50cc1029819f        docksal-dns            0.22%               14.83MiB / 3.858GiB   0.38%               74.9kB / 127kB      9.63MB / 0B         9
340be2a85d41        docksal-vhost-proxy    0.25%               20.36MiB / 3.858GiB   0.52%               6.29MB / 10.2MB     15.9MB / 123kB      11

  ```

</details>

**Comment by lpeabody** [comment]
I believe this is the commit that caused the regression https://github.com/docksal/docksal/commit/22c041bafe1fcef7c762f826cbf9d835460a10eb#diff-41adca34c3eac75ca9453c7cb0634f6a

**Comment by lpeabody** [comment]
If you want to leave this as is, update the **Overriding default host value** section to say something like `NGROK_HEADER=example.com fin share` instead of `fin share --host=example.com`.

**Comment by lpeabody** [comment]
Accidentally closed issue whoops :)

**Comment by lpeabody** [comment]
Thanks for confirming @sean-e-dietrich. I think updating the documentation section with my note above should be sufficient.

**Comment by achekulaev** [comment]
Closed in #604
